resources/lib/plugin.py

```diff
@@ -120,10 +186,168 @@ def play():
+    cat_id = plugin.args["id"][0]
+    cat_name = plugin.args["name"][0]
+    
+    try:
+        req = requests.get("https://www.rtp.pt/play/bg_l_pg/?listcategory={}&page={}".format(
+            cat_id,
+            page), headers=HEADERS).text
+    except:
+        raise_notification()
+
+    pagei = ListItem("[B]{}[/B] - Página {}".format(cat_name, page))
+    pagei.setProperty('IsPlayable', 'false')
+    addDirectoryItem(handle=plugin.handle, listitem=pagei, isFolder=False, url="")
+
+    soup = BeautifulSoup(req, 'html.parser')
+
+    for a in soup.find_all('a'):
+        url = a.get('href')
+        title = a.get('title')
+        img = a.find('img').get('src')
+        metas = a.find_next_sibling('i').find_all('meta')
+        description = metas[1].get('content')
+        ep = metas[0].get('content')[-12:]
+        
+        liz = ListItem("{} ({})".format(
+            kodiutils.compat_py23str(title),
+            kodiutils.compat_py23str(ep))
+        )
+        liz.setArt({"thumb": img,
+                    "icon": img,
+                    "fanart": kodiutils.FANART})
+        liz.setInfo("Video", infoLabels={"plot": kodiutils.compat_py23str(description), "title": kodiutils.compat_py23str(title)})
+
+        addDirectoryItem(
+            plugin.handle,
+            plugin.url_for(
+                programs_episodes,
+                title=kodiutils.compat_py23str(title),
+                ep=kodiutils.compat_py23str(ep),
+                img=kodiutils.compat_py23str(img),
+                description=kodiutils.compat_py23str(description),
+                url=kodiutils.compat_py23str(url),
+                page=1
+            ), liz, True)
+
+    newpage = str(int(page) + 1)
+    nextpage = ListItem("[B]{}[/B] - Página {} >>>".format(cat_name, newpage))
+    nextpage.setProperty('IsPlayable', 'false')
+    addDirectoryItem(handle=plugin.handle, listitem=nextpage, isFolder=True, url=plugin.url_for(programs_category, name=cat_name, id=cat_id, page=newpage))
+
+    endOfDirectory(plugin.handle)
+
+
+@plugin.route('/programs/episodes')
+def programs_episodes():
+    title = plugin.args["title"][0]
+    ep = plugin.args["ep"][0]
+    img = plugin.args["img"][0]
+    url = plugin.args["url"][0]
+    page = plugin.args["page"][0]
+
+    prog_id = url.split("/")[2][1:]
+
+    try:
+        req = requests.get("https://www.rtp.pt/play/bg_l_ep/?listProgram={}&page={}".format(
+            prog_id,
+            page), headers=HEADERS).text
+    except:
+        raise_notification()
+
+    pagei = ListItem("[B]{}[/B] - Página {}".format(title, page))
+    pagei.setProperty('IsPlayable', 'false')
+    addDirectoryItem(handle=plugin.handle, listitem=pagei, isFolder=False, url="")
+
+    soup = BeautifulSoup(req, 'html.parser')
+
+    for a in soup.find_all('a'):
+        url = a.get('href')
+        img = ""
+        if a.find('script') != None:
+            img = re.search(r'\'(.+?)\'', a.find('script').text).group(1)
```

Comment:
Try/except?

Reply:
isto n é algo q impeça a reprodução, simplesmente ha programas q por alguma razão não têm imagem lá no site

Reply:
A minha questão era mais relacionada com o group(0) que não tenho a certeza se é seguro, isto é, se verifica condições de out of bounds. Se a Regex não funcionar estás a apanhar um grupo inválido. Daí sugerir envolventes a chamada num try/except. Ou então verificares primeiro se o grupo existe

Reply:
na boa, adicionei um check pelo sim pelo nao